src/components/tl/ucp/tl_ucp_sendrecv.h

```diff
@@ -338,8 +443,15 @@ static inline ucc_status_t ucc_tl_ucp_put_nb(void *buffer, void *target,
         return status;
     }
 
+    if (src_memh) {
+        status = ucc_tl_ucp_get_memh(team, src_memh, &ucp_memh);
+        if (ucc_unlikely(UCC_OK != status)) {
+            return status;
+        }
+    }
+
     status = ucc_tl_ucp_resolve_p2p_by_va(team, target, &ep, dest_group_rank,
-                                          &rva, &rkey, &segment);
+                                          &rva, &rkey, &segment, dest_memh);
     if (ucc_unlikely(UCC_OK != status)) {
         return status;
```

Comment:
we probably need to cleanup here as well?

Reply:
I'm not sure there is anything to clean up. If there's an error in `ucc_tl_ucp_resolve_p2p_by_va`, it occurs before any allocations or unpacking of rkeys. please correct me if i'm wrong.